cedar-policy-validator/src/err.rs

```diff
@@ -140,28 +141,28 @@ pub enum SchemaError {
     #[diagnostic(help(
         "any entity types appearing anywhere in a schema need to be declared in `entityTypes`"
     ))]
-    UndeclaredEntityTypes(HashSet<String>),
```

Comment:
Can we `pub use` this `SchemaError` and then make it the only `SchemaError` struct?

Reply:
Do you recall why we needed this wrapper type originally?

Reply:
IIRC, it was because the internal enum variants with things like `UndeclaredENtityTypes(HashSet<Name>)` where `Name` was the internal type. This problem shouldn't exist with the opaque struct pattern we're using.

Reply:
Given that semver-check can't check re-exports of another crate (see https://github.com/obi1kenobi/cargo-semver-checks/issues/2), I think we'd better keep this type to avoid nasty semver issues.

Reply:
E.g., there could an internal struct `UndelaredEntityType` that implements `Diagnostic` and defines the error message text. Then we have the internal `SchemaError` struct hold that with `diagnostic(transparent)` and the external `SchemaError` holds an external wrapper struct `UndeclaredEntityType` that holds the internal `UndeclaredEntityStruct`, both with `diagnostic(transparant)`.

Reply:
Please check the latest changes, @john-h-kastner-aws